p2p/src/message.rs

```diff
@@ -104,16 +122,78 @@ impl BlockListResponse {
     }
 }
 
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct AddrListResponse {
+    addresses: Vec<PeerAddress>,
+}
+
+impl AddrListResponse {
+    pub fn new(addresses: Vec<PeerAddress>) -> Self {
+        Self { addresses }
+    }
+
+    pub fn addresses(&self) -> &Vec<PeerAddress> {
```

Comment:
Nitpick:
```suggestion
    pub fn addresses(&self) -> &[PeerAddress] {
```
I think that returning a slice is more idiomatic in such cases.

Reply:
Sure, updated. I used `Vec` for consistency with other messages.